src/core/qgsfield.cpp

```diff
@@ -268,6 +278,12 @@ QString QgsField::displayString( const QVariant &v ) const
   {
     return QObject::tr( "BLOB" );
   }
+  else if ( d->type == QVariant::StringList )
+  {
+    QString formatString = arrayFormatString() == QgsField::FormatHstore ? "{%1}" : "[%1]";
+    return formatString.arg( v.toStringList().join( "," ) );
```

Comment:
What happens if an element of the string list contains commas?

Reply:
I'll add a test to check it works